src/Evolution/Systems/Cce/LinearSolve.hpp

```diff
@@ -0,0 +1,204 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/ApplyMatrices.hpp"
+#include "NumericalAlgorithms/LinearOperators/IndefiniteIntegral.hpp"
+#include "NumericalAlgorithms/LinearOperators/Transpose.hpp"
+#include "NumericalAlgorithms/LinearSolver/Lapack.hpp"
```

Comment:
you can remove most of these includes now that you have moved the definitions to the cpp file, and just forward-declare what you need (I think fwd-declare `ComplexDataVector` + `Matrix` and include `Tensor/TypeAliases.hpp` should suffice).